server/controllers/controller-category.js

```diff
@@ -0,0 +1,200 @@
+                        })        
+                }else {
+                    res.status(500).json({ error : 'Maaf categories list tidak ditemukan!'})    
+                }  
+            })
+            .catch( error => {
+                res.status(500).json( { message : 'Error while update categories ', error:error.message})
+            })
+        
+    },
+    destroy : (req, res) => {
+        let id = ObjectId( req.params.id )
+        Category
+            .findByIdAndDelete( id )
+            .then( response => {
+                res.status(200).json( response )
+            })
+            .catch( error => {
+                res.status(500).json( { message : 'Error while destroy categories', error : error.message})
+            })
+    },
+    addItem : (req, res) => {
+        
+        console.log(req)
+        // let createdNewItem = null
+        // let category_id = req.body.category_id
+
+        // let item = new Item({ name : req.body.name, description : req.body.description, stock : Number(req.body.stock), price : Number(req.body.price ),url : req.file.cloudStoragePublicUrl})
+        
+        // item
+        //     .save()      
+        //     .then( newitem => {
+        //         createdNewItem = newitem
+        //         return Category.findById( category_id )
+        //     })
+        //     .then( category => {
+                
+        //         category.item_list.push(createdNewItem._id)
+        //         return category.save()
+        //     })
+        //     .then( response_adding_item_to_category => {
+        //         res.status(200).json( createdNewItem )
+        //     })
+        //     .catch( error => {
+        //         res.status(500).json({ message : error.message})
+        //     })
+    },
+    removeItem : (req, res) => {
+        let item_id = ObjectId( req.params.id )
+        let category_id = req.body.category_id
+        
+        Category
+            .findById( category_id )
+            .then( category => {
+                let item = category.item_list.find( item => {
+                    return item.equals( item_id )
+                })
+                res.send(item)
+                category.item_list.pull(item._id)
+                return category.save()
+            })
+            .then( response_remove_item_from_category => {
+                return Item.findByIdAndDelete(item_id)
+            })
+            .then( itemdeleted => {
+                res.status(200).json(itemdeleted)
+            })
+            .catch( error => {
+                res.status(500).json( { message : 'Error while remove item in categories', error : error.message})
+            })
+    },
+    addToCart : (req, res) =>{
+        
+        let buyer = null
+        let user_id = req._id
+        let item_id = req.params.id
+        let numberBuy = Number(req.body.numberBuy)
+
+        
+        User
```

Comment:
Kaya agak aneh aja kalau misal di controller category kita ngelakukan add to cart. Lebih enak ini dibuat controller sendiri untuk nge handle cart ini.